### ISSUE TITLE
ORA-01861 when casting string as date

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7434]
When casting an ISO formatted date string literal as a date in Oracle, we're getting

> ORA-01861: literal does not match format string

For example:

```sql
select (trunc(sysdate) - cast('2000-01-01' as date))
from dual
```

As parsed from:

```sql
SELECT current_date - '2000-01-01'::date
```

Is incorrect. In Oracle, unfortunately, the date format depends on the NLS settings of the system / session. We should use `TO_DATE()` instead. The same thing happens when casting a string as timestamp

**Comment by lukaseder** [comment]
The workaround is to use a plain SQL function call to ` TO_DATE(xyz, 'YYYY-MM-DD')`

**Comment by lukaseder** [comment]
This is fixed in jOOQ 3.12. A backport will not be scheduled, as the fix that generates a `TO_DATE()` or `TO_TIMESTAMP()` call might behave subtly differently from the `CAST()`, which may introduce new regressions.
